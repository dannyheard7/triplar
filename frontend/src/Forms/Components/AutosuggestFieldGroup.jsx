import React from "react";
import FormFieldGroup from "./FormFieldGroup";
import Autosuggest from "react-autosuggest";

export default class AutosuggestFieldGroup extends FormFieldGroup {
    renderInputComponent = inputProps => (
      <div className="form-group">
          <label>{this.props.label}</label>
          <input {...inputProps} className="form-control" name={this.props.name} required/>
            {this.props.help && <small className="form-text text-muted">{this.props.help}</small>}
            {this.props.errors &&
                <li className="list-unstyled">
                    {this.props.errors.map(this.createErrorItem)}
                </li>
            }
      </div>
    );

    onChange(event, { newValue }) {

    }

    render() {
        const inputProps = {
          value:  this.props.value,
          onChange: this.props.onChange,
        };

        return (
            <Autosuggest
                    suggestions={this.props.suggestions}
                    onSuggestionsFetchRequested={this.props.getSuggestions}
                    onSuggestionsClearRequested={this.props.onSuggestionsClearRequested}
                    getSuggestionValue={this.props.getSuggestionValue} renderSuggestion={this.props.renderSuggestion}
                    inputProps={inputProps}
                    shouldRenderSuggestions={this.props.shouldRenderSuggestions}
                    renderInputComponent={this.renderInputComponent}
                  />
        );
    }
}

AutosuggestFieldGroup.defaultProps = {
    value: "",
    suggestions: []
};