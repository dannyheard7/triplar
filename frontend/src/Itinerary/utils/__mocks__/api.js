const faker = require('faker');

export default {
    curryAPIFunc (apiFunc, id) {return (object) => apiFunc(id, object)},

    searchCities(value) {
        return Promise.resolve({status: 200, data: {data: {cities: [faker.address.city(),  faker.address.city()]}}});
    },

    addLocationToTrip(tripId, value) {
        return Promise.resolve({status: 200, data: value});
    }
}