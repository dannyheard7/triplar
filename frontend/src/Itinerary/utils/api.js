import axios from "axios";
import Q from "q";
import {backendHost} from 'App/utils/constants';

export default {
    curryAPIFunc (apiFunc, id) {return (object) => apiFunc(id, object)},

    addLocationToTrip(tripId, destination) {
        const mutateTripLocationQuery = ` 
            mutation AddTripLocation($input: TripLocationMutationInput!){
              result: addTripLocation(input: $input) {
                tripLocation {
                    id
                    startDate
                    endDate
                    city {
                        name
                        location
                    }
                    trip {
                        id
                    }
                }
                errors {
                  field
                  messages
                }
              }
            }`;


        let variables = {input: {trip: tripId, ...destination}};

        return Q.when(axios.post(`${backendHost}/api/graphql`, {query: mutateTripLocationQuery, variables: variables}));
    },

    getLocationDayItinerary(tripLocationId, date) {
        const getLocationDayItinerary = ` 
           query { 
            locationDayItinerary(date: "${date}", locationId: ${tripLocationId}) {
                place {
                    id
                    name
                    imageUrl
                }
            }
           }`;

        return Q.when(axios.post(`${backendHost}/api/graphql`, {query: getLocationDayItinerary}));
    },

    addItemToTripLocation(tripLocationId, placeId, date, order) {
        const addTripLocationItemMutation = ` 
            mutation AddTripLocationItem($input: AddTripLocationItemMutationInput!){
              addTripLocationItem(input: $input) {
                tripLocationItem {
                  order
                }
                errors {
                  field
                  messages
                }
              }
            }`;

        let variables = {input: {location: tripLocationId, apiPlace: placeId, order: order, startDate: date, endDate: date}};

        return Q.when(axios.post(`${backendHost}/api/graphql`, {query: addTripLocationItemMutation, variables: variables}));
    },

    searchCities(query) {
        const searchCities = ` 
           query { 
            cities(name: "${query}") {
                name
                country {
                    name
                }
            }
           }`;

        return Q.when(axios.post(`${backendHost}/api/graphql`, {query: searchCities}));
    },
}