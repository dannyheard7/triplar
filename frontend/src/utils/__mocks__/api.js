const faker = require('faker');

export default {
    curryAPIFunc (apiFunc, id) {return (object) => apiFunc(id, object)},

    registerUser(email, password, firstName, lastName) {
        return new Promise((resolve, reject) => {
           let response = {status: 201, data:  {'user': faker.internet.email()}}
           resolve(response);
        });
    },

    getLoginToken(email, password) {
        return new Promise((resolve, reject) => {
            let response = {status: 200, data: {'user': faker.internet.email(), 'token': faker.random.number() }};
            resolve(response);
        })
    },

    getTrips() {
        return new Promise((resolve, reject) => {
            let trips = [faker.random.word(), faker.random.word()];
            let response = {status: 200, data: trips};
            resolve(response);
        })
    },

    createTrip(trip) {
        return new Promise((resolve, reject) => {
            let response = {status: 201, data: trip};
            resolve(response)
        })
    },

    patchTrip(trip) {
        return new Promise((resolve, reject) => {
            let response = {status: 201, data: trip};
            resolve(response)
        })
    },

    getTrip(id)  {
         const trip = {
            'id': id, 'name': faker.random.word(), 'created_by': faker.internet.email()
        };
        return new Promise((resolve, reject) => {
            let response = {status: 200, data: trip};
            resolve(response);
        });
    },

    deleteTrip(tripId) {
        return new Promise((resolve, reject) => {
            let response = {status: 204};
            resolve(response)
        })
    },

    searchCities(value) {
        return new Promise((resolve, reject) => {
            let response = {status: 200, data: [faker.address.city(),  faker.address.city()]};
            resolve(response);
        });
    },

    addDestinationToTrip(tripId, value) {
        return new Promise((resolve, reject) => {
            let response = {status: 200, data: value};
            resolve(response);
        });
    }
}