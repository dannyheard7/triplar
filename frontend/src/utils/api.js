import axios from "axios";
import Q from "q";

export default {
    curryAPIFunc (apiFunc, id) {return (object) => apiFunc(id, object)},

    registerUser(user){
        return Q.when(axios.post("/api/register/", user));
    },

    getLoginToken(user) {
        return Q.when(axios.post("/api/token/", user));
    },

    getTrips() {
        return Q.when(axios.get("/api/trips/"));
    },

    createTrip(trip) {
        return Q.when(axios.post("/api/trips/", trip));
    },

    getTrip(tripId) {
        return Q.when(axios.get("/api/trips/" + tripId));
    },

    patchTrip(tripId, trip) {
        return Q.when(axios.patch("/api/trips/" + tripId, trip));
    },

    deleteTrip(tripId) {
        return Q.when(axios.delete("/api/trips/" + tripId));
    },

    addDestinationToTrip(tripId, destination) {
         return Q.when(axios.post("/api/trips/" + tripId + "/destination", destination));
    },

    searchCities(query) {
        return Q.when(axios.get("/api/places/cities/search?query=" + query));
    }
}