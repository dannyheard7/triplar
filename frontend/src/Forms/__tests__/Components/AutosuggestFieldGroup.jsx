import React from "react";
import {shallow} from "enzyme";
import api from "utils/api.js";

import AutosuggestFieldGroup from "Forms/Components/AutosuggestFieldGroup.jsx";

const faker = require('faker');
jest.mock('utils/api.js');

describe('<AutosuggestFieldGroup />', () => {
    test.skip('renderInputComponent ', () => {})

});