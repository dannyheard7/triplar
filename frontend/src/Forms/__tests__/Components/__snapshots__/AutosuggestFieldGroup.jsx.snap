// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AutosuggestFieldGroup /> inputComponent renders correctly 1`] = `
<div
  className="form-group"
>
  <label />
  <input
    className="form-control"
    errors={
      Array [
        "Error",
      ]
    }
    label="City Search"
    name={undefined}
    required={true}
  />
</div>
`;
